src/transforms/transaction.rs

```diff
@@ -0,0 +1,812 @@
+use super::Transform;
+use crate::{
+    conditions::{AnyCondition, Condition},
+    event::discriminant::Discriminant,
+    event::{Event, LogEvent, Value},
+    topology::config::{DataType, TransformConfig, TransformContext, TransformDescription},
+};
+use async_stream::stream;
+use bytes::{Bytes, BytesMut};
+use chrono::{DateTime, Utc};
+use futures::{
+    compat::{Compat, Compat01As03},
+    stream,
+    stream::StreamExt,
+};
+use futures01::Stream as Stream01;
+use indexmap::IndexMap;
+use serde::{Deserialize, Serialize};
+use std::collections::{hash_map, HashMap};
+use std::time::{Duration, Instant};
+use string_cache::DefaultAtom as Atom;
+
+//------------------------------------------------------------------------------
+
+#[derive(Deserialize, Serialize, Debug, Default)]
+#[serde(deny_unknown_fields, default)]
+pub struct TransactionConfig {
+    pub expire_after_ms: Option<u64>,
+
+    pub flush_period_ms: Option<u64>,
+
+    /// An ordered list of fields to distinguish transactions by. Each
+    /// transaction has a separate event merging state.
+    #[serde(default)]
+    pub identifier_fields: Vec<String>,
```

Comment:
So it seems like `[]` is valid and the default here? I see it's used in the `Discriminant`. Could you document what reasons someone might want to use `[]`?